src/v/kafka/server/quota_manager.h

```diff
@@ -73,12 +71,13 @@ class quota_manager : public ss::peering_sharded_service<quota_manager> {
         std::optional<atomic_token_bucket> pm_rate;
     };
 
-    using client_quotas_map_t
-      = absl::node_hash_map<tracker_key, ss::lw_shared_ptr<client_quota>>;
-    using client_quotas_t = ssx::sharded_ptr<client_quotas_map_t>;
+    using local_map_t
+      = chunked_hash_map<tracker_key, std::shared_ptr<client_quota>>;
+
+    using global_map_t
+      = chunked_hash_map<tracker_key, std::shared_ptr<client_quota>>;
```

Comment:
@pgellert where is the use of `std::shared_ptr` being documented? there is probably what like 1 or 2 uses in the entire tree, and its always discouraged. this use case is fine, but how will future developers learn about the usage?

a good place for it would be in the code as a comment, or in the commit message, but i don't see any information in either place.

can you please point me at the documentaiton or submit a pr that adds it?

Reply:
Makes sense. I'm adding it now here: https://github.com/redpanda-data/redpanda/pull/20802

Reply:
thanks @pgellert :)